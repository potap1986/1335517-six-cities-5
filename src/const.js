export const Sorting = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const ReviewLength = {
  MIN: 50,
  MAX: 300,
};

export const MAX_RATE = 5;
export const DEFAULT_HOVER = -1;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  OFFER: `/offer/:id`,
  MAIN_PAGE: `/`,
  FAVORITES: `/favorites`,
  LOGIN: `/login`,
};

export const APIRoute = {
  HOTELS: `/hotels`,
  HOTEL: `/hotels/`,
  LOGIN: `/login`,
  REVIEWS: `/comments`,
  FAVORITE: `/favorite`,
};

export const HttpCode = {
  UNAUTHORIZED: 401,
  SUCCESS: 200
};

export const CITIES = [`Amsterdam`, `Paris`, `Cologne`, `Brussels`, `Hamburg`, `Dusseldorf`];
