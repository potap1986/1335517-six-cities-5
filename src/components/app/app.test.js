import React from 'react';
import {Provider} from 'react-redux';
import renderer from "react-test-renderer";
import App from './app';
import configureStore from 'redux-mock-store';
import {CITIES, Sorting, AuthorizationStatus} from '../../const';

jest.mock(`../private-route/private-route`, () => `PrivateRoute`);
jest.mock(`../main-page/main-page`, () => `MainPage`);
jest.mock(`../login-screen/login-screen`, () => `LoginScreen`);
jest.mock(`../favorites-screen/favorites-screen`, () => `FavoritesScreen`);
jest.mock(`../offer-screen/offer-screen`, () => `OfferScreen`);

const storeConfig = configureStore();

const store = storeConfig({
  DATA: {
    cities: CITIES,
    offers: null,
    nearOffers: null,
    reviews: null,
    offer: null,
  },
  APPLICATION: {
    hoveredOffer: -1,
    currentCity: CITIES[0],
    offersForCity: [],
    sortType: Sorting.POPULAR,
  },
  USER: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    user: {
      id: 0,
      email: ``,
      name: ``,
      avatar: ``,
      isPro: false,
    }
  },
});

describe(`Render App`, () => {
  it(`Render App`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <App />
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
