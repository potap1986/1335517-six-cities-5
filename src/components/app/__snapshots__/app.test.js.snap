// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Render App 1`] = `
<MainPage
  onOfferClick={[Function]}
/>
`;
